File: frontend/src/constants/matchCancelReason.js
// src/constants/matchCancelReason.js

export const MatchCancelReason = {
    NOT_ENOUGH_PLAYERS: "NOT_ENOUGH_PLAYERS",
    TECHNICAL_ISSUE: "TECHNICAL_ISSUE",
    WEATHER: "WEATHER",
    ORGANIZER_DECISION: "ORGANIZER_DECISION",
    OTHER: "OTHER",
};

export const MATCH_CANCEL_REASON_OPTIONS = [
    {
        value: MatchCancelReason.NOT_ENOUGH_PLAYERS,
        label: "Nedostatečný počet hráčů",
    },
    {
        value: MatchCancelReason.TECHNICAL_ISSUE,
        label: "Technické problémy (led, hala, doprava)",
    },
    {
        value: MatchCancelReason.WEATHER,
        label: "Nepříznivé počasí",
    },
    {
        value: MatchCancelReason.ORGANIZER_DECISION,
        label: "Rozhodnutí organizátora",
    },
    {
        value: MatchCancelReason.OTHER,
        label: "Jiný důvod",
    },
];

File: backend/src/main/java/cz/phsoft/hokej/match/enums/MatchCancelReason.java
package cz.phsoft.hokej.match.enums;

/**
 * Výčtový typ reprezentující důvody zrušení zápasu.
 *
 * Používá se ve vazbě na MatchStatus pro jednoznačné
 * a strojově zpracovatelné určení příčiny zrušení zápasu.
 * Hodnota je ukládána do databáze jako textový enum.
 */
public enum MatchCancelReason {

    /**
     * Nedostatečný počet přihlášených hráčů.
     */
    NOT_ENOUGH_PLAYERS,

    /**
     * Technické problémy, například led, hala nebo doprava.
     */
    TECHNICAL_ISSUE,

    /**
     * Nepříznivé počasí.
     */
    WEATHER,

    /**
     * Rozhodnutí organizátora.
     */
    ORGANIZER_DECISION,

    /**
     * Jiný, blíže nespecifikovaný důvod.
     */
    OTHER
}